dev/io.openliberty.microprofile.health.3.1.internal/src/io/openliberty/microprofile/health31/services/impl/HealthCheck31ExecutorImpl.java

```diff
@@ -0,0 +1,155 @@
+
+import com.ibm.websphere.csi.J2EEName;
+import com.ibm.websphere.csi.J2EENameFactory;
+import com.ibm.ws.javaee.version.JavaEEVersion;
+import com.ibm.ws.microprofile.health.services.HealthCheckBeanCallException;
+import com.ibm.ws.microprofile.health.services.impl.AppModuleContextService;
+import com.ibm.wsspi.threadcontext.WSContextService;
+
+import io.openliberty.microprofile.health31.services.HealthCheck31CDIBeanInvoker;
+import io.openliberty.microprofile.health31.services.HealthCheck31Executor;
+
+@Component(service = HealthCheck31Executor.class, configurationPolicy = ConfigurationPolicy.IGNORE, property = { "service.vendor=IBM" })
+public class HealthCheck31ExecutorImpl implements HealthCheck31Executor {
+
+    /**
+     * For retrieving the CMD and classloader context for a given
+     * app/module name.
+     */
+    private AppModuleContextService appModuleContextService;
+
+    /**
+     * Jakarta EE version if Jakarta EE 9 or higher. If 0, assume a lesser EE spec version.
+     */
+    private volatile int eeVersion;
+
+    /**
+     * Tracks the most recently bound EE version service reference. Only use this within the set/unsetEEVersion methods.
+     */
+    private ServiceReference<JavaEEVersion> eeVersionRef;
+
+    private HealthCheck31CDIBeanInvoker healthCheckCDIBeanInvoker;
+    private J2EENameFactory j2eeNameFactory;
+    private final static Logger logger = Logger.getLogger(HealthCheck31ExecutorImpl.class.getName(), "io.openliberty.microprofile.health31.resources.Health31");
+
+    private final static String HC_MANAGEDTASK_IDENTITY_NAME = "mp.healthcheck.proxy";
+    private final static String HC_TASK_OWNER = "mp.healthcheck.runtime";
+    private final static String ONLY_WAR_EJB_NOT_SUPPORTED = null;
+
+    @Reference
+    protected void setAppModuleContextService(AppModuleContextService appModuleContextService) {
+        this.appModuleContextService = appModuleContextService;
+    }
+
+    @Reference
+    protected void setHealthCheckApplicationBeanInvoker(HealthCheck31CDIBeanInvoker healthCheckCDIBeanInvoker) {
+        this.healthCheckCDIBeanInvoker = healthCheckCDIBeanInvoker;
+    }
+
+    @Reference
+    protected void setJ2EENameFactory(J2EENameFactory j2eeNameFactory) {
+        this.j2eeNameFactory = j2eeNameFactory;
+    }
+
+    @Override
+    public Set<HealthCheckResponse> runHealthChecks(String appName, String moduleName, String healthCheckProcedure) throws HealthCheckBeanCallException {
+
+        J2EEName j2eeName = j2eeNameFactory.create(appName, moduleName, ONLY_WAR_EJB_NOT_SUPPORTED);
+
+        Map<String, String> execProps = new HashMap<String, String>();
+        Set<HealthCheckResponse> retval;
+
+        // TaskIdentity identifies the task for the purposes of mgmt/auditing.
+        final String MANAGEDTASK_IDENTITY_NAME = eeVersion < 9 ? "javax.enterprise.concurrent.IDENTITY_NAME" : "jakarta.enterprise.concurrent.IDENTITY_NAME";
```

Comment:
you don't need this check. MP Health 3.1 doesn't support EE 9 ... it isn't possible to start this feature at the same time as EE9 features.

Reply:
Updated to remove the check for both 3.0 and 3.1.